### ISSUE TITLE
401 Unauthorized when using AWS ECR image registry

### ISSUE THREAD

**Issue by AlessioFranceschi** [testcontainers/testcontainers-node#752]
**Expected Behaviour**
When using a third party registry using the new `TESTCONTAINERS_HUB_IMAGE_NAME_PREFIX` env variable, I expect it to pull the image correctly if i'm logged in. 
I'm using AWS ECR, in the terminal I can run the following commands:
```bash
aws ecr get-login-password --region <AWS_REGION> | docker login --username AWS --password-stdin <AWS_ACCOUNT>.dkr.ecr.<AWS_REGION>.amazonaws.com
```
```bash
docker pull image <my_image>
```

And it correctly pulls the image.

**Actual Behaviour**
1. I set the env variable `TESTCONTAINERS_HUB_IMAGE_NAME_PREFIX` to `<AWS_ACCOUNT>.dkr.ecr.<AWS_REGION>.amazonaws.com`
2. I try to create a container:
```js
const container = await new GenericContainer('myContainer:latest')
            .withPullPolicy(PullPolicy.alwaysPull())
            .start();
```
3. When running, I get this error: 
```Error: (HTTP code 500) server error - failed to resolve reference "<AWS_ACCOUNT>.dkr.ecr.<AWS_REGION>.amazonaws.com/myContainer:latest": unexpected status from HEAD request to <AWS_ACCOUNT>.dkr.ecr.<AWS_REGION>.amazonaws.com/v2/myContainer/manifests/latest: 401 Unauthorized```

**Testcontainer Logs**
```
Error: (HTTP code 500) server error - failed to resolve reference "<AWS_ACCOUNT>.dkr.ecr.<AWS_REGION>.amazonaws.com/myContainer:latest": unexpected status from HEAD request to <AWS_ACCOUNT>.dkr.ecr.<AWS_REGION>.amazonaws.com/v2/myContainer/manifests/latest: 401 Unauthorized
    at /Users/user/Code/repo/node_modules/.pnpm/docker-modem@3.0.8/node_modules/docker-modem/lib/modem.js:343:17
    at IncomingMessage.<anonymous> (/Users/user/Code/repo/node_modules/.pnpm/docker-modem@3.0.8/node_modules/docker-modem/lib/modem.js:370:9)
    at IncomingMessage.emit (node:events:531:35)
    at endReadableNT (node:internal/streams/readable:1696:12)
    at processTicksAndRejections (node:internal/process/task_queues:82:21)
```

**Steps to Reproduce**
1. Set `TESTCONTAINERS_HUB_IMAGE_NAME_PREFIX` to an AWS ECR url
2. Create a container with an image from that registry

**Environment Information**
- Operating System: MacOS Sonoma 14.4.1
- Docker Version: 4.28.0 
- Node version: v21.7.1
- Testcontainers version: 10.8.1

**Comment by cristianrgreco** [comment]
Fixed in https://github.com/testcontainers/testcontainers-node/pull/748. Will be released shortly

**Comment by cristianrgreco** [comment]
Fixed in 10.8.2 (hopefully)
